fuzzers/fuzzbench_text/src/lib.rs

```diff
@@ -61,6 +62,121 @@ use libafl_targets::{
 };
 #[cfg(unix)]
 use nix::{self, unistd::dup};
+use serde::{Deserialize, Serialize};
+
+#[derive(Default, Serialize, Deserialize, Clone, Debug)]
+pub struct FuzzbenchDumpMetadata {
+    pub last_queue: usize,
+    pub last_crash: usize,
+}
+
+libafl::impl_serdeany!(FuzzbenchDumpMetadata);
+
+pub struct FuzzbenchDumpStage<CB, EM, Z> {
+    crashes_dir: PathBuf,
+    queue_dir: PathBuf,
+    to_bytes: CB,
+    phantom: PhantomData<(EM, Z)>,
+}
+
+impl<CB, EM, Z> UsesState for FuzzbenchDumpStage<CB, EM, Z>
+where
+    EM: UsesState,
+{
+    type State = EM::State;
+}
+
+impl<CB, E, EM, Z> Stage<E, EM, Z> for FuzzbenchDumpStage<CB, EM, Z>
+where
+    CB: FnMut(&<Z::State as UsesInput>::Input) -> Vec<u8>,
+    EM: UsesState<State = Z::State>,
+    E: UsesState<State = Z::State>,
+    Z: UsesState,
+    Z::State: HasCorpus + HasSolutions + HasRand + HasMetadata,
+{
+    #[inline]
+    fn perform(
+        &mut self,
+        _fuzzer: &mut Z,
+        _executor: &mut E,
+        state: &mut Z::State,
+        _manager: &mut EM,
+        _corpus_idx: usize,
+    ) -> Result<(), Error> {
+        let meta = state
+            .metadata()
+            .get::<FuzzbenchDumpMetadata>()
+            .map(|m| m.clone())
+            .unwrap_or_else(|| FuzzbenchDumpMetadata::default());
+
+        let queue_count = state.corpus().count();
+        let crashes_count = state.solutions().count();
+
+        for i in meta.last_queue..queue_count {
+            let mut testcase = state.corpus().get(i)?.borrow_mut();
+            let input = testcase.load_input()?;
+            let bytes = (self.to_bytes)(input);
+
+            let fname = self.queue_dir.join(format!("id_{}", i));
+            let mut f = File::create(fname).expect("Unable to open file");
+            drop(f.write(&bytes));
+        }
+
+        for i in meta.last_crash..crashes_count {
+            let mut testcase = state.solutions().get(i)?.borrow_mut();
+            let input = testcase.load_input()?;
+            let bytes = (self.to_bytes)(input);
+
+            let fname = self.crashes_dir.join(format!("id_{}", i));
+            let mut f = File::create(fname).expect("Unable to open file");
+            drop(f.write(&bytes));
```

Comment:
I think there's a better method, like `write_all`, that you want to use

Reply:
there's no difference

Reply:
Of course there is. Write:
```This function will attempt to write the entire contents of buf, but the entire write might not succeed, or the write may also generate an error. A call to write represents at most one attempt to write to any wrapped object.

Calls to write are not guaranteed to block waiting for data to be written, and a write which would otherwise block can be indicated through an [Err](https://doc.rust-lang.org/std/result/enum.Result.html#variant.Err) variant```

Write_all:
```Attempts to write an entire buffer into this writer.

This method will continuously call [write](https://doc.rust-lang.org/std/io/trait.Write.html#tymethod.write) until there is no more data to be written or an error of non-[ErrorKind::Interrupted](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.Interrupted) kind is returned. This method will not return until the entire buffer has been successfully written or such an error occurs. The first error that is not of [ErrorKind::Interrupted](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.Interrupted) kind generated from this method will be returned.```

Reply:
The experiment started, I don't think will make any difference

Reply:
Fix it anyway for future reference